src/components/parameters/ParameterBaseString.svelte

```diff
@@ -17,28 +17,46 @@
   export let levelPadding: number = 20;
   export let parameterType: ParameterType = 'activity';
   export let use: ActionArray = [];
+  export let type: string = 'string';
 
   const dispatch = createEventDispatcher<{
     change: FormParameter;
     reset: FormParameter;
   }>();
 
   $: columns = `calc(${labelColumnWidth}px - ${level * levelPadding}px) auto`;
+
+  function handleChange(): void {
+    dispatch('change', formParameter);
+  }
 </script>
 
 <div class="parameter-base-string" style="grid-template-columns: {columns}">
   <ParameterName {formParameter} />
   <Input>
-    <input
-      bind:value={formParameter.value}
-      class="st-input w-full"
-      class:error={formParameter.errors !== null}
-      aria-label={formParameter.name}
-      {disabled}
-      type="text"
-      use:useActions={use}
-      on:change={() => dispatch('change', formParameter)}
-    />
+    {#if type === 'password'}
+      <input
+        bind:value={formParameter.value}
+        class="st-input w-full"
+        class:error={formParameter.errors !== null}
+        aria-label={formParameter.name}
+        {disabled}
+        type="password"
+        use:useActions={use}
+        on:change={handleChange}
+      />
+    {:else}
+      <input
+        bind:value={formParameter.value}
+        class="st-input w-full"
+        class:error={formParameter.errors !== null}
+        aria-label={formParameter.name}
+        {disabled}
+        type="string"
```

Comment:
Not necessarily relevant if you implement my suggestion above, but the value of `type` for strings is "text"